src/sc3_array.c

```diff
@@ -406,13 +407,16 @@ sc3_array_new_view (sc3_allocator_t * alloc, sc3_array_t ** view,
   /* create array and adjust for being an array view */
   SC3E (sc3_array_new (alloc, view));
   (*view)->esize = a->esize;
-  (*view)->ealloc = a->ealloc - offset;
   (*view)->ecount = length;
   (*view)->mem = a->mem + (*view)->esize * offset;
 
   (*view)->viewed = a;
   SC3E (sc3_array_ref (a));
 
+  lg = sc3_log2_ceil (length, SC3_INT_BITS - 1);
+  SC3A_CHECK (0 <= lg && lg < SC3_INT_BITS - 1);
+  SC3A_CHECK (length <= (1 << lg));
+  (*view)->ealloc = 1 << lg;
```

Comment:
I think this power of 2 requirement is articial/historic and not useful for views (especially those on data). Let's just assign ealloc = length for both view constructions and update the test in sc3_array_is_valid for views accordingly.

P.S. Thanks for the tests!